SignalMessaging/categories/Promise+retainUntilComplete.swift

```diff
@@ -14,6 +14,7 @@ public extension AnyPromise {
         // compiler warning: "Variable 'retainCycle' was written to, but never read"
         var retainCycle: AnyPromise? = self
         self.always {
+            assert(retainCycle != nil)
```

Comment:
I think it'll be nice to assert this - and it has the side effect of fixing the "variable assigned but never read" warning.